int32 currNode
bool currType
int32[2] chosenEdge
